zio/bzngio/writer.go

```diff
@@ -4,25 +4,43 @@ import (
 	"io"
 
 	"github.com/brimsec/zq/zcode"
+	"github.com/brimsec/zq/zio"
 	"github.com/brimsec/zq/zng"
 	"github.com/brimsec/zq/zng/resolver"
 )
 
 type Writer struct {
 	io.Writer
-	encoder *resolver.Encoder
-	buffer  []byte
+	zio.Flags
+	encoder      *resolver.Encoder
+	buffer       []byte
+	frameRecords int
 }
 
-func NewWriter(w io.Writer) *Writer {
+func NewWriter(w io.Writer, flags zio.Flags) *Writer {
 	return &Writer{
-		Writer:  w,
-		encoder: resolver.NewEncoder(),
-		buffer:  make([]byte, 0, 128),
+		Writer:       w,
+		Flags:        flags,
+		encoder:      resolver.NewEncoder(),
+		buffer:       make([]byte, 0, 128),
+		frameRecords: -1,
 	}
 }
 
+func (w *Writer) NewFrame() error {
```

Comment:
Nit: Maybe call this StartFrame or StartNewFrame since NewFrame reads like a constructor.